### DISCUSSION TITLE
Add remote-powered "Notices" to the top of the homescreen

### DISCUSSION THREAD

**Discussion by hawkrives** [StoDevX/AAO-React-Native#5545] [Ideas]
I'm thinking something like

> All About Olaf is a student- and alumni-developed application; it is not sponsored by the College in any way, shape, or form. Please direct bug reports to us by way of the "Contact Us" button in the app settings.

---

A future idea is to fetch this from GitHub, so that if the app ever dies we can note that it's unmaintained.

**Comment by drewvolz** [comment]
I get where you're coming from. However, we specify this in quite a few places as it is:

1. App description as the _first_ thing you read about the [App Store](https://itunes.apple.com/us/app/all-about-olaf-sga/id938588319?mt=8) and [Google Play](https://play.google.com/store/apps/details?id=com.allaboutolaf&hl=en) apps.

> (This app is created, supported, and managed by the St. Olaf College Student Government Association (SGA); this is not officially developed by staff of St. Olaf College.)

2. [Legal view](https://github.com/StoDevX/AAO-React-Native/blob/master/data/legal.md)

> This software is in no way supported by or sponsored by St. Olaf College.
The college shall in no way be responsible for the support of this application.

3. [Credits view](https://github.com/StoDevX/AAO-React-Native/blob/master/data/credits.yaml#L11-L13)

> The app is supported as well as published by the Student Government Association (SGA) web team, called the Oleville Development Team.

<hr>

I _really_ don't think this needs to be a top-level string of information. I just don't feel we need to always show that information off. I wouldn't object to a one-time-only screen or alert that describes this on install that forces the user to acknowledge this.

**Comment by hawkrives** [comment]
I think that it should be on the homescreen, where it's visible and remotely updated so that when the apps finally die, we can do one last thing to let everyone know that they're unmaintained. That's why I want it at the top level.

**Comment by drewvolz** [comment]
Just trigger a view or alert once you update that remote data. It's the same concept... all that is changing is that the app is checking `maintained = false` which would do [anything else] other than what it does normally.

Displaying the 39 word / 212 character message all the time is something I have never seen an app do. It's odd to display the support status on the first screen constantly. That feels alien to me.

**Comment by hawkrives** [comment]
Most apps that you use are also the official app of whatever service you use; I feel like we're in a bit of a special place here.

**Comment by hawkrives** [comment]
Being used by so many people and yet not supported by the organizations.

**Comment by drewvolz** [comment]
Special circumstances can warrant a change, sure. But I'd prefer to leave the homescreen alone?

**Comment by hawkrives** [comment]
I'll just adopt the new global alert system from the Olaf website, then? 😛 

![screen shot 2017-09-18 at 4 36 33 pm](https://user-images.githubusercontent.com/464441/30565896-8bec15f8-9c8f-11e7-8a09-7440b882f566.png)

**Comment by drewvolz** [comment]
I would like to hear from @rye @elijahverdoorn @mattk410 @erichkauffman on this

**Comment by rye** [comment]
I think that promoting a notice like this (especially one of such length) to the _Home_ screen is a bit excessive.  I'd be totally fine with putting it in at the bottom of the Settings view, though.

As far as putting in a dead-man's-switch of sorts—I'd totally be fine with doing _something_ like that, but I don't think that we should have this be always-displayed info.  It's totally fine by me to have the app phone home or build in some kind of outdated version checker, but to add such a notice to the bottom... unless we're getting complaints from users or we witness users complaining about it, I think it's pretty clear from the fact that SGA is marketing it that it's SGA's thing and not any part of the college.

But I'm open to counterarguments.

**Comment by hawkrives** [comment]
I actually have a follow up proposal after thinking about this some more:

What if we added a triggerable alert view at the top of the homescreen, between then navigation and the icons?

It could conceivably be used for everything from "elections are today" (maybe) to "campus lockdown" to "app is no longer maintained", powered by the presence of a json file on github pages - probably looking like `{severity: info | alert | notice, message: string, title?: string}` - which I'm imagining to take after the new Olaf website's alerts, as seen above. 

This way, it covers my main concern for the future (I might bring up the "unofficial" thing again, but put it in settings, we'll see, I've dropped it for now) and also covers new use cases that we don't have a story for yet.

**Comment by rye** [comment]
I like this idea, and it's a deeper take on the original idea, providing some flexibility.  Check out [react-native-dropdownalert](https://www.npmjs.com/package/react-native-dropdownalert) which looks like a well-designed implementation of a dropdown alert-y thingy.

Also, something something push notifications?

**Comment by hawkrives** [comment]
I'll make a branch for a prototype after we cut 2.4, I think. Thanks for the discussion!

**Comment by elijahverdoorn** [comment]
Good thoughts all around. I like the drop down alert ideas, this whole discussion reminds me to look at push notifications again....

**Comment by drewvolz** [comment]
> @hawkrives comment [from above](https://github.com/StoDevX/AAO-React-Native/issues/1650#issuecomment-330426050)
> What if we added a triggerable alert view at the top of the homescreen, between then navigation and the icons?
> 
> It could conceivably be used for everything from "elections are today" (maybe) to "campus lockdown" to "app is no longer maintained", powered by the presence of a json file on github pages - probably looking like {severity: info | alert | notice, message: string, title?: string} - which I'm imagining to take after the new Olaf website's alerts, as seen above.
> 
> This way, it covers my main concern for the future (I might bring up the "unofficial" thing again, but put it in settings, we'll see, I've dropped it for now) and also covers new use cases that we don't have a story for yet.

This seems more relevant these days with our constant hours switching and closing of things due to snow.
